olp-cpp-sdk-dataservice-read/src/repositories/QuadTreeIndex.cpp

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "QuadTreeIndex.h"
+
+#include <algorithm>
+#include <bitset>
+#include <iostream>
+
+#include <rapidjson/document.h>
+#include <rapidjson/istreamwrapper.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+constexpr auto kParentQuadsKey = "parentQuads";
+constexpr auto kSubQuadsKey = "subQuads";
+constexpr auto kDataHandleKey = "dataHandle";
+constexpr auto kVersionKey = "version";
+constexpr auto kSubQuadKeyKey = "subQuadKey";
+constexpr auto kPartitionKey = "partition";
+
+QuadTreeIndex::QuadTreeIndex() {}
+
+QuadTreeIndex::QuadTreeIndex(BlobDataPtr data) {
+  if (data == nullptr || data->empty()) {
+    return;
+  }
+  data_ = reinterpret_cast<DataHeader*>(data->data());
+  size_ = data->size();
+}
+
+QuadTreeIndex::QuadTreeIndex(const olp::geo::TileKey& root, int depth,
```

Comment:
Another option for this would be to use the JsonParser.h structure. Can be done in a later phase.